3rdparty/openenclave/ert.patch

```diff
@@ -138,6 +138,57 @@ index df4adc056..2c04586cf 100644
  set(CMAKE_CXX_STANDARD_REQUIRED ON)
  # Do not use, for example, `-std=gnu++14`.
  set(CMAKE_CXX_EXTENSIONS OFF)
+diff --git a/common/sgx/collateral.c b/common/sgx/collateral.c
+index f4d9fcc76..81f4f22aa 100644
+--- a/common/sgx/collateral.c
++++ b/common/sgx/collateral.c
+@@ -326,18 +326,44 @@ oe_result_t oe_validate_revocation_list(
+                 "Failed to read CRL. %s",
+                 oe_result_str(result));
+         }
+-        // Otherwise, CRL should have v3 structure which is DER encoded
++        // Otherwise, CRL should have v3 or v3.1 structure
++        //    v3 is hex encoded DER
++        //    v3.1 is DER
+         else
+         {
+             size_t der_data_size = sgx_endorsements->items[i].size;
+             if (der_data_size == 0)
+                 OE_RAISE(OE_INVALID_PARAMETER);
+
+-            // If DER CRL buffer has null terminator, need to remove it before
++            // If CRL buffer has null terminator, need to remove it before
+             // send the DER data to crypto API for reading.
+             if (sgx_endorsements->items[i].data[der_data_size - 1] == 0)
+                 der_data_size -= 1;
+
++            // If the CRL is only composed of hex digits we need to manually convert to DER
++            bool ishex = true;
++            for (size_t l = 0; l < der_data_size; ++l)
++            {
++                uint8_t c = sgx_endorsements->items[i].data[l];
++                if (!((c >= 'A' && c <= 'F') || (c >= 'a' && c <= 'f') || (c >= '0' && c <= '9')))
++                {
++                    ishex = false;
++                    break;
++                }
++            }
++            if (ishex)
++            {
++                char *tmp = (char*)oe_malloc(der_data_size);
```

Comment:
tmp leaks if oe_hex_to_buf fails. I think you don't need to malloc and can use oe_hex_to_buf in-place.